packages/doenetml-worker/lib-doenetml-core/src/dast/ref_resolve.rs

```diff
@@ -169,6 +171,56 @@ impl Resolver {
         }
     }
 
+    /// Delete `nodes` from the resolver
+    pub fn delete_nodes(&mut self, nodes: &[FlatNode]) {
+        for node in nodes.iter() {
+            if let FlatNode::Element(element) = node {
+                if let Some(name) = get_element_name(element) {
+                    let mut parent_option = self.node_parent[node.idx()];
+
+                    while let Some(parent_idx) = parent_option {
+                        let name_map = &mut self.name_map[parent_idx];
+
+                        let references = name_map.remove(&name);
+
+                        match references {
+                            Some(Ref::Unique(idx)) => {
+                                // if the index doesn't match the deleted node, add it back to the name map
```

Comment:
When would we expect this to happen?

Reply:
I expanded the comment to address this question.